platform/ios/src/MGLMapView.mm

```diff
@@ -1533,17 +1560,25 @@ - (void)handleTwoFingerTapGesture:(UITapGestureRecognizer *)twoFingerTap
     }
     else if (twoFingerTap.state == UIGestureRecognizerStateEnded)
     {
-        CGPoint gesturePoint = [self anchorPointForGesture:twoFingerTap];
-
-        mbgl::ScreenCoordinate center(gesturePoint.x, gesturePoint.y);
-        _mbglMap->scaleBy(0.5, center, MGLDurationInSecondsFromTimeInterval(MGLAnimationDuration));
-
-        __weak MGLMapView *weakSelf = self;
-
-        [self animateWithDelay:MGLAnimationDuration animations:^
+        MGLMapCamera *oldCamera = self.camera;
+        if ([self.delegate respondsToSelector:@selector(mapView:shouldChangeFromCamera:toCamera:)]
+            && ![self.delegate mapView:self shouldChangeFromCamera:oldCamera toCamera:self.camera])
```

Comment:
self.camera at this point is equal to oldCamera. We need to pass in the destination camera as the third parameter. Could you create a copy of the camera and modify it to reflect the result after zooming out?